### PR TITLE
Transfer restrictions and Genesis Account Ownership Transfer

### PR DESCRIPTION
**Pull request by libermans**
### 🎯 Overview

This PR implements two production-ready systems for professional blockchain bootstrap management:

1. **Transfer Restrictions System** - Temporarily blocks user-to-user transfers during bootstrap while preserving essential operations (gas, staking, governance). Enables coordinated exchange launches with 90-day default window.

2. **Genesis Account Ownership Transfer System** - Enables secure, atomic transfer of complete genesis account ownership including all liquid balances and vesting schedules. Designed for clean handover of pre-allocated accounts to team members, advisors, and institutional partners.

#### Transfer Restrictions
- **Selective Blocking**: Blocks user-to-user transfers, preserves gas/staking/governance operations
- **Emergency Exemptions**: Governance-approved transfer templates with usage limits
- **Auto-Cleanup**: Restrictions lift automatically at configured block height with zero overhead
- **Default Configuration**: 0 blocks (testing/testnet), 1,555,000 blocks (~90 days) for production via genesis

#### Genesis Account Ownership
- **Complete Transfer**: Atomic transfer of liquid balances + all vesting schedules with timeline preservation
- **Security Controls**: One-time enforcement, optional whitelist, signature-based authorization
- **Vesting Support**: All Cosmos SDK vesting types (Periodic, Continuous, Delayed, Base)
- **Audit Trail**: Complete transfer records with historical tracking and compliance reporting

#### Module Architecture
```
x/restrictions/          x/genesistransfer/
├── keeper/             ├── keeper/
├── types/             ├── types/
├── module/            ├── module/
└── testutil/          └── testutil/
```

#### Core Logic
**Transfer Restrictions**: SendRestriction function intercepts bank transfers, applies validation logic, handles emergency exemptions with automatic unregistration at deadline.

**Genesis Transfer**: Unified `TransferOwnership()` function performs atomic balance + vesting transfer with comprehensive validation, transfer record management, and recipient account creation.

#### Testing Coverage
- **500+ Unit Tests**: Core logic, edge cases, error conditions
- **Integration Tests**: Real keeper setup with mock dependencies  
- **E2E Tests**: Complete workflow testing (`RestrictionsTests.kt`, `GenesisTransferTests.kt`)

#### Documentation
- **Module READMEs**: Architecture, integration patterns, API reference
- **Deployment Guides**: Production procedures, genesis configuration, monitoring setup
- **CLI Guides**: Complete command reference with security best practices

#### Core Implementation
- `inference-chain/x/restrictions/` - Transfer restrictions module
- `inference-chain/x/genesistransfer/` - Genesis account ownership module
- `inference-chain/proto/inference/` - Protobuf definitions for both modules
- `inference-chain/app/` - Module registration with dependency injection

#### API & Testing
- `decentralized-api/internal/server/` - REST API handlers for restrictions
- `testermint/src/test/kotlin/` - Complete E2E test suites for both modules
- `testermint/src/main/kotlin/` - CLI integration and data classes

#### Documentation
- `proposals/` - Design documents and implementation plans
- `docs/specs/` - Complete operational documentation
- Module READMEs with architecture and usage guides

### CODE REVIEW

File: docs/specs/genesistransfer/genesistransfer-cli-guide.md [hunk 1]
```diff
@@ -0,0 +1,630 @@
...
+```json
+{
+  "is_transferred": true,
+  "transfer_record": {
+    "genesis_address": "cosmos1genesis123example456",
+    "recipient_address": "cosmos1recipient789example",
+    "transfer_height": "1234567",
+    "completed": true,
+    "transferred_denoms": ["nicoin"],
+    "transfer_amount": "1000000000"
```
Comment (DimaOrekhovPS) [inline docs/specs/genesistransfer/genesistransfer-cli-guide.md]: Are we transferring all the funds a genesis account had? It's more about account ownership than money, right?

File: docs/specs/genesistransfer/genesistransfer-deployment.md [hunk 2]
```diff
@@ -0,0 +1,738 @@
...
+  
+  echo "---"
+done
+```
+
+## Security Considerations and Private Key Management
+
+### 1. Authority Key Security
+
+The genesis transfer module requires governance authority for execution:
```
Comment (DimaOrekhovPS) [inline docs/specs/genesistransfer/genesistransfer-deployment.md]: Does it mean that participants will vote for each transfer?

File: docs/specs/genesistransfer/genesistransfer-cli-guide.md (hunk 1 above)
Comment (libermans) [reply docs/specs/genesistransfer/genesistransfer-cli-guide.md]: we should transfer full sum, yes

File: docs/specs/genesistransfer/genesistransfer-deployment.md (hunk 2 above)
Comment (libermans) [reply docs/specs/genesistransfer/genesistransfer-deployment.md]: Sorry, this documentation is not what I intended, will read it through and change

File: inference-chain/x/genesistransfer/keeper/transfer.go [hunk 3]
```diff
@@ -0,0 +1,684 @@
...
+		// No vesting account, nothing to transfer
+		return nil
+	}
+
+	if err != nil {
+		return errorsmod.Wrapf(err, "failed to create vesting account")
+	}
+
+	// Set the new vesting account for the recipient
+	k.accountKeeper.SetAccount(ctx, newVestingAccount)
```
Comment (DimaOrekhovPS) [inline inference-chain/x/genesistransfer/keeper/transfer.go:134]: Are we deleting the old account record? Belonging to the gensesis pre-transfer account?

**Review by DimaOrekhovPS — APPROVED** [review]

File: decentralized-api/internal/server/public/restrictions_handlers.go
Comment (gmorgachev) [inline decentralized-api/internal/server/public/restrictions_handlers.go:1]: it's okay now but i'm still thinking that proxying requests to chain data via api is not the best practice

**Review by gmorgachev — APPROVED** [review]
